### DISCUSSION TITLE
[FR] Climate sub-button automatic background color

### DISCUSSION THREAD

**Discussion by AndLindemann** [Clooos/Bubble-Card#979] [Feature requests]
For consistency, I believe it'd be great if a "climate" sub-button would change its background color automatically based on the climate mode, in the same way as the background color option for a climate card does it for the whole card:

Like here, the climate card is light-blue because it is in "cool" mode, but the same entity used in a sub-button is just showing the standard "active" color background. Would be nice if we'd have the option to let the button change its color based on the state as well, using the same colors as the climate type card.

![climate](https://github.com/user-attachments/assets/017df942-ffa5-4ce6-a965-c0705b6fb0c6)

**Comment by ChipSkylark37** [comment]
EDIT: Moved into new discussion: https://github.com/Clooos/Bubble-Card/discussions/984

I have a similar request, that I may put into a separate discussion.

Would it be possible that when the system is in AC / Heat mode (for a Heat Pump system) that it could color the entire card with either a 'blue' or 'orange' color to indicate the mode its in? Idle would be the generic color, and then if its in AC, it switches to color the entire card BLUE, or ORANGE if in heat mode.

I looked through some of the 'style' settings but don't know if this is possible.

**Comment by cb12tre** [comment]
I'm joining this request as well. It would be really nice to have the colors aligned for both "card_type climate" and "card_type button" when the sub_button is of type climate.
It would also be great to be able to customize the colors based on the "hvac_action" state.
